packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2824,4 +2826,24 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.state.isDeclareContext = oldIsDeclareContext;
       }
     }
+
+    // `...args` can have a negative impact on performance.
+    parseMethod(...args: any[]) {
+      const method = super.parseMethod(...args);
+      if ((method: any).abstract) {
+        const hasBody =
+          // For estree plugin
+          method.type === "MethodDefinition"
```

Comment:
We can do `this.hasPlugin("estree")` in this case, so that it's self-explanatory.